File: 0226-invert-binary-tree/0226-invert-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
    void dfs(TreeNode* root) {
        if (!root) return;
        if (root->left) dfs(root->left);
        if (root->right) dfs(root->right);
        swap(root->left, root->right);
    }
public:
    TreeNode* invertTree(TreeNode* root) {
        TreeNode* curr = root;
        dfs(curr);
        return root;
    }
};

File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        def dfs(curr):
            if (curr == None): return
            if (curr.left != None): dfs(curr.left)
            if (curr.right != None): dfs(curr.right)
            curr.left, curr.right = curr.right, curr.left

        dfs(root)
        return root
        